mpi-proxy-split/test/run_mana_mpi_regression_test.py

```diff
@@ -0,0 +1,202 @@
+import sys
+import os
+import subprocess
+import time
+import argparse
+from threading import Thread
+
+from subprocess import PIPE
+
+verbose = False
+
+# Wrapper to eprint to stderr
+def eprint(verbose, args):
+    if verbose:
+        print(args, file=sys.stderr)
+
+# Custom argparser to eprint error message
+class CustomParser(argparse.ArgumentParser):
+    def error(self, message):
+        sys.stderr.write('error: %s\n' % message)
+        self.print_help()
+        os._exit(2)
+
+# Error handling thread for timeouts
+def timeout_exit(timeout):
+    time.sleep(int(timeout))
+    sys.stderr.write('ERROR: TEST FAILED BY TIMEOUT\n')
+    sys.stderr.flush()
+    os._exit(11)
+
+# Remove unnecessary files (old checkpoint files)
+def clean():
+    subprocess.run('rm -rf ckpt_rank_*', shell=True)
+    subprocess.run('rm -rf dmtcp_restart_*', shell=True)
+
+# Helper function to get the number of currently running ranks
+def get_running_ranks(mana_dir, state=None):
+    status = subprocess.run([f"{mana_dir}/bin/mana_status"], stdout=PIPE,
+                             stderr=subprocess.DEVNULL)
+    if state is None:
+        return status.stdout.decode('utf-8').count(".mana.exe")
+    else:
+        return status.stdout.decode('utf-8').count(state)
+
+'''
+Example usage:
+
+python3 run_mana_mpi_regression_test.py send_recv_loop -i 12 -n 3 -m ~/mana
+
+'''
+def main():
+    # Parse arguments
+    parser = CustomParser(description='Run a MANA Test')
+    parser.add_argument('-i', '--iterations', metavar='I',
+                         help='Number of iterations for test', required=True)
+    parser.add_argument('test', metavar='T', help='Path to test case to run')
+    parser.add_argument('-n','--num_ranks', metavar='N', help='Number of ranks\
+                              for test', required=True)
+    parser.add_argument('-m','--mana_root', metavar='M', help='Absolute \
```

Comment:
Since these scripts are part of the MANA source, can we pick `MANA_ROOT` from the absolute path name of this script in case someone doesn't provide the `-m` option?